flink/v1.14/flink/src/test/java/org/apache/iceberg/flink/TestFlinkUpsert.java

```diff
@@ -203,40 +220,40 @@ public void testPrimaryKeyFieldsAtBeginningOfSchema() {
     LocalDate dt = LocalDate.of(2022, 3, 1);
     try {
       sql(
-          "CREATE TABLE %s(data STRING NOT NULL, dt DATE NOT NULL, id INT, PRIMARY KEY(data,dt) NOT ENFORCED) "
-              + "PARTITIONED BY (data) WITH %s",
+          "CREATE TABLE %s(id INT, name STRING NOT NULL, dt DATE NOT NULL, PRIMARY KEY(name,dt) NOT ENFORCED) "
```

Comment:
this test is for primary key at beginning of schema. this changed the order of the fields. we probably should do `dt, name, id`

Reply:
Thanks for your suggestion, my understanding of this test has improved.
 I have re-submitted the code and you may review again to check what else needs to be changed when you are free.